spec/community.mediawiki

```diff
@@ -55,3 +55,29 @@ of any unfilled orders.
 
 Less drastic punitive measures such as a cool-down period may be considered
 for minor, first-time or infrequent conduct violations.
+
+===Penalization Notification===
+
+In the event that a rule is broken, the client will be sent a notification that
+includes the broken rule, a detailed description of the offense, the time of
+the penalization, and the duration. If the client is offline, this information
+can be found in the [[comm.mediawiki/#Session_Authentication|connect response]].
+
+'''Notification route:''' <code>penalize</code>, '''originator: ''' dex
+
+<code>payload</code>
+{|
+! field      !! type   !! description
+|-
+| accountid   || string || hex encoding of the client's account ID
+|-
+| broken rule || int || the rule that was broken
+|-
+| timestamp   || int || the UNIX time in milliseconds at which the penalization occured
+|-
+| duration    || int || the duration in milliseconds of the penalization
+|-
+| details     || string || a message describing the penalization
+|-
+| sig         || string || hex-encoded server's signature of the serialized penalization data
+|}
```

Comment:
In a strikes system, a violation will not necessarily result in account suspension, but we still want to notify the user.  We could add a `strikes` field as a placeholder now or add it later when we get there.  If the violation does not start a suspension, then `duration` could be 0.

Reply:
Would different penalties come with different number of strikes?

So, the first strike might not even come with trading suspension. I hadn't thought of that.

Reply:
I would actually like to add strikes later, as you suggest, as I think it needs a little more conversation. We need to decide how many strikes per broken rule, lifetime of strikes, duration of penalty depending on strikes, and how many strikes until you're out.

Reply:
That's fine, we'll work towards a more gradual system.

I think strikes could be used in two ways: (1) trading suspension when a strikes threshold is reached, and (2) strikes influence the duration of the suspension.  Note that a suspended account that is still negotiating swaps can accumulate further strikes if they walk away from these swaps.